### PR TITLE
Feature/scalar multiplication

### PR DESCRIPTION
**Pull request by simonkerscher**
This adds multiplication by 1-dimensional vectors (with a physical quantity) to the vector class.

This allows calls like 
```
Vector<PhysicalType::Acceleration, double, 3> acceleration;
Vector<PhysicalType::Mass, double, 1> mass;
Vector<PhysicalType::Force, double, 3> force = mass * acceleration;
```
with physical-type checking.

whereas we currently need
```
Vector<PhysicalType::Acceleration, double, 3> acceleration;
Vector<PhysicalType::Mass, double, 1> mass;
Vector<PhysicalType::Force, double, 3> force(mass.toImplementation() * acceleration.toImplementation());
```

or more commonly use
```
Vector<PhysicalType::Acceleration, double, 3> acceleration;
double mass;
Vector<PhysicalType::Force, double, 3> force(mass * acceleration.toImplementation());
```

### CODE REVIEW

**Comment by remod** [comment]
Thanks for the contribution!

The changes look good to me. Does it make sense to add another test to check for 1D * 1D and 1D / 1D calculations? They seem to be a corner case, testing them would make sure all the compile time magic works as expected :)

**Comment by simonkerscher** [comment]
> Thanks for the contribution!
> 
> The changes look good to me. Does it make sense to add another test to check for 1D * 1D and 1D / 1D calculations? They seem to be a corner case, testing them would make sure all the compile time magic works as expected :)

Yes, 1*1 operations need to be tested.
This is covered in test case 

```
TEST_F(ScalarTest, multiplyTwoScalars) {
  kindr::Acceleration<double, 1> accelerationScalar(2.);
  kindr::Force<double, 1> forceScalar(10.);

  EXPECT_EQ(accelerationScalar * mass, forceScalar);
  EXPECT_EQ(mass * accelerationScalar, forceScalar);
  EXPECT_EQ(forceScalar / mass, accelerationScalar);
  EXPECT_EQ(forceScalar / accelerationScalar, mass);
}
```

File: include/kindr/vectors/Vector.hpp [hunk 1]
```diff
@@ -395,6 +437,16 @@ class Vector : public VectorBase<Vector<PhysicalType_, PrimType_, Dimension_> >,
     return *this;
   }
 
+  /*! \brief Multiplication with a scalar and assignment
+   * @param factor  factor with physical type
+   * @returns reference
+   */
+  template <PhysicalType PhysicalTypeFactor_, typename PrimTypeFactor_>
+  Vector<PhysicalType_, PrimType_, Dimension_>& operator*=(Vector<PhysicalTypeFactor_, PrimTypeFactor_, 1> factor) {
```
Comment (remod) [inline include/kindr/vectors/Vector.hpp:445]: Shouldn't `*=` and `/=` only work if the factor is of physical type `Typeless`? Otherwise we would need to change the type of `this`, right? (As this PR is already imported into our internal mono-repo, just let me know your opinion, and I will do potential adaptations internally.)

**Comment by remod** [comment]
Merged with https://github.com/ANYbotics/kindr/commit/fcea3c9bb9d7af93c6a781288a2ec0bf334c0d92, with the following adaptations:

* split up the traits for vector operations - now the traits and macros to define the return types of operations on physical quantities are independent of the dimensions of the vectors
* `operator*=` and `operator/=` for 1D vectors now re-use the respective version for scalars.
* add typedefs for the physical quantities time, mass, and intertia
* fixed many copy-paste mistakes in the description of physical quantities typedefs
